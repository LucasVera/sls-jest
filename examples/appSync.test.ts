const template = `
#set($id=$ctx.args.id)
{
  "id": "$id"
}
`;

describe('Mapping Template', () => {
  it('should evaluate a template', async () => {
    const expected = `
{
  "id": "123"
}
`;
    await expect({
      template,
      context: {
        arguments: {
          id: '123',
        },
      },
    }).toEvaluateTo(expected);
  });

  it('should evaluate a template snapshot', async () => {
    await expect({
      template,
      context: {
        arguments: {
          id: '456',
        },
      },
    }).toEvaluateToSnapshot();
  });

  it('should evaluate a template inline snapshot', async () => {
    await expect({
      template,
      context: {
        arguments: {
          id: '789',
        },
      },
    }).toEvaluateToInlineSnapshot(`
      "
      {
        \\"id\\": \\"789\\"
      }
      "
    `);
  });
});
