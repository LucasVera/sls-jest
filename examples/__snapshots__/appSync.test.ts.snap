// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mapping Template should evaluate a template snapshot 1`] = `
"
{
  \\"id\\": \\"456\\"
}
"
`;
